src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -632,11 +662,308 @@ internal static unsafe bool TryParseInt32(ReadOnlySpan<char> s, NumberStyles sty
             return true;
         }
 
+        /// <summary>Parses int limited to styles that make up NumberStyles.Integer.</summary>
+        private static bool TryParseInt32IntegerStyle(ReadOnlySpan<char> source, NumberStyles styles, NumberFormatInfo info, out int result, out bool overflow)
+        {
+            overflow = false;
+            Debug.Assert((styles & ~NumberStyles.Integer) == 0);
```

Comment:
This assert seems kind of weird. We assert that only the `NumberStyles.Integer` bit is set, but then branch below for various other bits. Doesn't seem to be a valid assert.

Reply:
NumberStyles.Integer isn't a single bit; it's defined as:
```C#
Integer = AllowLeadingWhite | AllowTrailingWhite | AllowLeadingSign,
```

Reply:
Oops. Sorry.